core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringParser.java

```diff
@@ -109,12 +109,13 @@ public QueryBuilder fromXContent(QueryParseContext parseContext) throws IOExcept
                             fField = parser.text();
                         }
 
+                        //norelease we need to delay access to the mapper service to the shard
```

Comment:
I think we should fix this now given that this query is already refactored, this seems like a bug